modules/server_processes_manager/php/process.class.inc

```diff
@@ -152,7 +152,7 @@ class Process
         // exit code in $_exitCodeFile
         $this->_pid = intval(
             exec(
-                "(bash -c '$mainCmd ; $getExitCodeCmd') > /dev/null 2> /dev/null & "
```

Comment:
neither of these are good, it should be using https://www.php.net/manual/en/function.escapeshellarg.php

Reply:
`$mainCmd` has already been shell-escaped earlier: escaping it twice would break the command. As for `$getExitCodeCmd`, it is generated in this class and is shell safe.

Reply:
Where is `$mainCmd` being shell escaped? I just expanded the diff for more context and it looks like it's just passed in from the constructor

Reply:
It is, but the command passed to the constructor is always defined in a class extending `AbstractServerProcess` and the shell escaping is done directly in the derived class (see for example `modules/server_processes_manager/php/mriuploadserverprocess.class.inc`) so the command is _already_ protected against shell injection.